### PR TITLE
Adding Pilot service annotation capabilities

### PR DESCRIPTION
**Pull request by knabben**
**Please provide a description of this PR:**

A few monitoring clouds (i.e DataDog) read from the Service annotation to gather metrics from, this PR adds Service annotation from `pilot` so `istiod` can be integrated into the monitoring system.

```
metadata:
    annotations:
        ad.datadoghq.com/endpoints.check_names: '["istio"]'
        ad.datadoghq.com/endpoints.init_configs: '[{}]'
        ad.datadoghq.com/endpoints.instances: |
            [
              {
                "istiod_endpoint": "http://%%host%%:15014/metrics",
                "send_histograms_buckets": true
              }
            ]
```

### CODE REVIEW

**Comment by knabben** [comment]
This can be overwritten directly in `operator` manifest though, can someone confirm operators should patch it instead of directly helm field change?

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    pilot:
      k8s:
        serviceAnnotations:
          "ad.datadog.com": "bla"
```

**Review by morvencao — COMMENTED** [review]
/ok-to-test

**Comment by jacob-delgado** [comment]
LGTM, but you need to run `make gen` first.

**Comment by knabben** [comment]
@jacob-delgado it's failing on `failed validation with: unknown field "serviceAnnotations" in v1alpha1.PilotConfig`.. 

The point is that the API does not allow it, and it's already being injected on `spec.components.pilot.k8s.serviceAnnotations`, making me believe the operator should be used instead.

**Comment by jacob-delgado** [comment]
Fixes https://github.com/istio/istio/issues/35229

**Review by jacob-delgado — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/templates/service.yaml [hunk 1]
```diff
@@ -3,6 +3,10 @@ kind: Service
 metadata:
   name: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}
   namespace: {{ .Release.Namespace }}
+  annotations:
+  {{- if .Values.pilot.serviceAnnotations }}
+{{ toYaml .Values.pilot.serviceAnnotations | indent 4 }}
+  {{- end }}
```
Comment (shamsher31) [inline manifests/charts/istio-control/istio-discovery/templates/service.yaml:9]: This avoids creating empty annotation [like](https://github.com/istio/istio/pull/35202/files#diff-5d70e6b65d86515abc1bcdf047a0cbbadff8858086ba2ea07ad85b4c8c274bdeR1416).
```suggestion
  {{- if .Values.pilot.serviceAnnotations }}
  annotations:
{{ toYaml .Values.pilot.serviceAnnotations | indent 4 }}
  {{- end }}
```

**Review by Monkeyanator — APPROVED** [review]

**Pull request merged**